src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -304,6 +305,12 @@ public bool IsSet(AvaloniaProperty property)
 
             VerifyAccess();
 
+            var scheduler = AvaloniaLocator.Current.GetService<IScheduler>();
+            if (scheduler != null)
+            {
+                source = source.ObserveOn(scheduler);
```

Comment:
There's a small problem here in that `GetDescription` will be called on the produced observable which won't retain the description. Binding descriptions are good for debugging so it would be good to keep them working. This can be fixed by taking the description before this line.